fix: BROS-71: Add playback cursor for timeseries sync

makseq opened the pull request:
This PR adds a playback cursor. 


https://github.com/user-attachments/assets/944f6375-147c-48dc-94df-511c01ad3240


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

makseq committed:
fix: BROS-71: Add playback cursor for timeseries sync

yyassi-heartex commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15420639371)

yyassi-heartex committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15420639371

makseq committed:
Fixes

makseq committed:
Merge branch 'fb-bros-71' of github.com:heartexlabs/label-studio into fb-bros-71

makseq committed:
Merge branch 'develop' of github.com:heartexlabs/label-studio into fb-bros-71

makseq committed:
Readme and fixes

robot-ci-heartex committed:
ci: Build tag docs

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15421675778

makseq committed:
Suppress overview sync

makseq committed:
Merge branch 'fb-bros-71' of github.com:heartexlabs/label-studio into fb-bros-71

makseq committed:
Readme adds

bmartel commented on web/libs/editor/src/tags/object/TimeSeries.jsx:
```diff
@@ -1068,6 +1109,17 @@ const Overview = observer(({ item, data, series }) => {
+      .attr("class", "overview-playhead")
+      .attr("y1", 0)
+      .attr("y2", focusHeight)
+      .attr("stroke", item.cursorcolor || "#1e90ff")
```
```suggestion
      .attr("stroke", item.cursorcolor || "var(--color-neutral-inverted-surface)")
```
@ricardoantoniocm what is the semantic default color we would use here for a playhead?

ricardoantoniocm replied:
We should use `--color-neutral-inverted-surface` in this instance.

This ensures consistency with the playhead of the audio player.
![image](https://github.com/user-attachments/assets/73a37662-315a-45e3-b53d-5d033c20dce3)

Red usually carries a negative/destructive connotation so we should avoid it unless there's a very good reason for it.

bmartel commented on web/libs/editor/src/tags/object/TimeSeries.jsx:
```diff
@@ -89,6 +90,9 @@ const TagAttrs = types.model({
   sync: types.maybeNull(types.string),
+
+  // Color of the playback cursors (hex or any SVG-compatible color string)
+  cursorcolor: types.optional(types.string, "#1e90ff"),
```
```suggestion
  cursorcolor: types.optional(types.string, "var(--color-neutral-inverted-surface)"),
```

@ricardoantoniocm what is the semantic default color we would use here for a playhead?

ricardoantoniocm replied:
We should use `--color-neutral-inverted-surface` in this instance.

bmartel commented on web/libs/editor/src/tags/object/TimeSeries/Channel.jsx:
```diff
@@ -837,6 +847,47 @@ class ChannelD3 extends React.Component {
+      .attr("class", "playhead")
+      .attr("y1", 0)
+      .attr("y2", this.height)
+      .attr("stroke", this.props.item.parent?.cursorcolor || "#1e90ff")
```
```suggestion
      .attr("stroke", this.props.item.parent?.cursorcolor || "var(--color-neutral-inverted-surface)")
```
@ricardoantoniocm here as well

ricardoantoniocm replied:
We should use `--color-neutral-inverted-surface` in this instance.

makseq committed:
Doc refinements

makseq committed:
Fix docs

cloudmark commented:
Spoke to @makseq and we agreed that we would tackle the dragging of the play heads later. This is so that this feature is in line with the other components like video and audio.  Great stuff @makseq , this was needed.

makseq committed:
Update web/libs/editor/src/tags/object/TimeSeries.jsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

makseq committed:
Update web/libs/editor/src/tags/object/TimeSeries.jsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

makseq committed:
Update web/libs/editor/src/tags/object/TimeSeries/Channel.jsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

makseq committed:
New cursor view. Docs fixes

makseq committed:
Docs

makseq committed:
Merge branch 'develop' of github.com:heartexlabs/label-studio into fb-bros-71

makseq committed:
Fix docs

makseq committed:
Merge branch 'develop' of github.com:heartexlabs/label-studio into fb-bros-71
